.circleci/jobs/deploy.yml

```diff
@@ -2,14 +2,25 @@
 
 parameters:
   hosts:
-    description: 'Hosts to deploy to'
+    description: "Hosts to deploy to"
     type: string
   service:
-    description: 'Service to deploy on hosts'
+    description: "Service to deploy on hosts"
     type: string
+  freeze:
+    description: "Whether to freeze and skip deploy"
+    type: string
+    default: "false"
 machine:
   image: ubuntu-2004:202201-02
 steps:
+  - run:
+      name: Is frozen
+      command: |
+        freeze=(<< parameters.freeze >>)
+        if [ "$freeze" = "true" ]; then
+          circleci-agent step halt
+        fi
```

Comment:
Could you use a circleci `- when:` for this instead?

Reply:
unfortunately not. see commit history. can't coerce the env var to a boolean 😫

Reply:
What about something like this? we do this in client

```
- when:
        condition:
          matches: { pattern: '^stage$', value: << parameters.build-type >> }
```